libs/common/src/middleware/front-end-error-handler.get-front-end-error-handler.test.ts

```diff
@@ -0,0 +1,72 @@
+import httpMocks from 'node-mocks-http';
+import { ErrorRequestHandler, Request, Response } from 'express';
+import { HttpStatusCode } from 'axios';
+import createHttpError from 'http-errors';
+import { getFrontEndErrorHandler } from './front-end-error-handler';
+
+describe('front-end-error-handler', () => {
+  describe('frontEndErrorHandler', () => {
+    console.error = jest.fn();
+
+    let frontEndErrorHandler: ErrorRequestHandler;
+    let res: httpMocks.MockResponse<Response>;
+    let req: httpMocks.MockRequest<Request>;
+    let next: jest.Mock;
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      ({ res, req } = httpMocks.createMocks());
+      next = jest.fn();
+    });
+
+    describe('when the error is a CSRF token error', () => {
+      beforeEach(() => {
+        frontEndErrorHandler = getFrontEndErrorHandler();
+      });
+
+      it('redirects the user to /', () => {
+        const error = createHttpError(HttpStatusCode.ImATeapot, { code: 'EBADCSRFTOKEN' });
+
+        frontEndErrorHandler(error, req, res, next);
+
+        expect(res._getRedirectUrl()).toEqual('/');
+        expect(res._getStatusCode()).toEqual(HttpStatusCode.Found);
+      });
+    });
+
+    describe('when the error is an unhandled error', () => {
+      describe('when the problem with service template is not provided', () => {
+        const defaultProblemWithServiceTemplateLocation = '_partials/problem-with-service.njk';
+        beforeEach(() => {
+          frontEndErrorHandler = getFrontEndErrorHandler();
```

Comment:
Could just have this in the `it` below instead of having a `beforeEach` for 1x test

Same in the next `describe`

Reply:
updated